File: longest-substring-wout-dups/longest-substring-wout-dups.py
def lengthOfLongestSubstring(self, s):
	hash_table = {}
	maxLength = 0
	start = 0

	for i in range(len(s)):

		if (s[i] in hash_table): 
			start = max(start, hash_table[s[i]]+1)

		maxLength = max(maxLength,i-start+1)
		hash_table[s[i]]=i

	return maxLength

File: longest-substring-wout-dups/longest-substring-wout-dups.js
/**
 * @param {string} s
 * @return {number}
 */
var lengthOfLongestSubstring = function(s) {
  let start = 0, res = 0;
  let charsInWindow = new Set();
  
  for (let end = 0; end < s.length; end++) {
      let entering = s[end];
      console.log('test', entering)
      
      while (charsInWindow.has(entering)) {
          charsInWindow.delete(s[start++]);
      }
      charsInWindow.add(s[end]);
      console.log('look', s[end])
      res = Math.max(res, end - start + 1);
      console.log(res, end - start + 1)
  }
  return res;
}